sceptre/template.py

```diff
@@ -288,12 +290,22 @@ def _render_jinja_template(template_dir, filename, jinja_vars):
         :returns: The body of the CloudFormation template.
         :rtype: str
         """
-        logger = logging.getLogger(__name__)
-        logger.debug("%s Rendering CloudFormation template", filename)
-        env = jinja2.Environment(
-            loader=jinja2.FileSystemLoader(template_dir),
-            undefined=jinja2.StrictUndefined
-        )
-        template = env.get_template(filename)
-        body = template.render(**jinja_vars)
-        return body
+        try:
+            logger = logging.getLogger(__name__)
+            logger.debug("%s Rendering CloudFormation template", filename)
+            env = jinja2.Environment(
+                loader=jinja2.FileSystemLoader(template_dir),
+                undefined=jinja2.StrictUndefined
+            )
+            template = env.get_template(filename)
+            body = template.render(**jinja_vars)
+            return body
+
+        except Exception as e:
+            _, _, tb = sys.exc_info()
+            stack = traceback.extract_tb(tb)
+            for frame in stack:
+                if re.match(r'.*\.j2', frame.filename):
+                    print("Jinja template error in {} - file {}, line {}:".format(frame.name, frame.filename, frame.lineno))
```

Comment:
We could probably use a `logger.exception` rather than a `print`. For the end use achieves the same thing.

Reply:
I'll change it. By the way I also had an idea about how to catch all template errors in general. Move the exception handling code to the calling function and display stack frames from any file within the project templates directory. What do you think?

Reply:
I have changed it around a bit. The exception is now handled in the parent function (`body` property) so that we get line numbers for troposphere templates also. When handling exceptions, any file which exists in the `templates` directory will be included in the printed stack trace. I used `logger.error` because `logger.exception` prints an actual stack trace to the terminal when used.